### PR TITLE
improve error hanlding of inplace update env from metadata

### PR DESCRIPTION
**Pull request by veophi**
Signed-off-by: mingzhou.swx <mingzhou.swx@alibaba-inc.com>

### Ⅰ. Describe what this PR does
Improve robustness for some codes

### CODE REVIEW

File: pkg/controller/containerrecreaterequest/crr_controller.go [hunk 1]
```diff
@@ -60,6 +61,10 @@ var (
 	controllerKind       = appsv1alpha1.SchemeGroupVersion.WithKind("ContainerRecreateRequest")
 )
 
+func newPodReadinessControl(adp podadapter.Adapter) utilpodreadiness.Interface {
+	return utilpodreadiness.NewForAdapter(adp)
```
Comment (zmberg) [inline pkg/controller/containerrecreaterequest/crr_controller.go:65]: Why re-define the func newPodReadinessControl, other than immediate call utilpodreadiness.NewForAdapter(adp)?

Comment (veophi) [reply pkg/controller/containerrecreaterequest/crr_controller.go]: done

**Pull request merged**